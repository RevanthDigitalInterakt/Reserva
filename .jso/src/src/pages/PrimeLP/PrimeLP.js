  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _asyncToGenerator2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _slicedToArray2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[2]));
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[3]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[4]);
  var _testProps = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[5]));
  var _PrimeBenefits = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[6]));
  var _PrimeFAQ = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[7]));
  var _PrimeHero = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[8]));
  var _PrimeIntro = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[9]));
  var _PrimeSubscribe = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[10]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function PrimeLP(_ref) {
    var navigation = _ref.navigation;
    var _usePrimeInfo = (0, _$$_REQUIRE(_dependencyMap[11]).usePrimeInfo)(),
      onAddPrimeToCart = _usePrimeInfo.onAddPrimeToCart,
      isPrime = _usePrimeInfo.isPrime;
    var _useApolloFetchPolicy = (0, _$$_REQUIRE(_dependencyMap[12]).useApolloFetchPolicyStore)(['getFetchPolicyPerKey']),
      getFetchPolicyPerKey = _useApolloFetchPolicy.getFetchPolicyPerKey;
    var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      loadingAddCartPrime = _useState2[0],
      setLoadingAddCartPrime = _useState2[1];
    var _usePrimeStore = (0, _$$_REQUIRE(_dependencyMap[13]).usePrimeStore)(['animationBag', 'handleClickContinue', 'isVisibleModalWelcome', 'changeStateAnimationBag', 'changeStateIsVisibleModalWelcome']),
      animationBag = _usePrimeStore.animationBag,
      handleClickContinue = _usePrimeStore.handleClickContinue,
      isVisibleModalWelcome = _usePrimeStore.isVisibleModalWelcome,
      changeStateAnimationBag = _usePrimeStore.changeStateAnimationBag,
      changeStateIsVisibleModalWelcome = _usePrimeStore.changeStateIsVisibleModalWelcome;
    var _useLandingPagePrimeQ = (0, _$$_REQUIRE(_dependencyMap[14]).useLandingPagePrimeQuery)({
        context: {
          clientName: 'gateway'
        },
        fetchPolicy: getFetchPolicyPerKey('landingPagePrime')
      }),
      rawData = _useLandingPagePrimeQ.data,
      loading = _useLandingPagePrimeQ.loading;
    var data = (0, _react.useMemo)(function () {
      return rawData == null ? undefined : rawData.landingPagePrime;
    }, [rawData == null ? undefined : rawData.landingPagePrime]);
    var handleOnModalHideSignIn = (0, _react.useCallback)(function () {
      changeStateAnimationBag(true);
    }, [changeStateAnimationBag]);
    var handleOnModalHide = (0, _react.useCallback)(function () {
      if (isPrime) {
        setTimeout(function () {
          changeStateIsVisibleModalWelcome(true);
        }, 500);
      }
    }, [changeStateIsVisibleModalWelcome, isPrime]);
    var onAddToCart = (0, _react.useCallback)(/*#__PURE__*/(0, _asyncToGenerator2.default)(function* () {
      try {
        if (!data || loadingAddCartPrime) return;
        if (isPrime) {
          navigation.popToTop();
          return;
        }
        setLoadingAddCartPrime(true);
        yield onAddPrimeToCart(false);
        handleOnModalHideSignIn();
      } catch (e) {
        _$$_REQUIRE(_dependencyMap[15]).ExceptionProvider.captureException(e);
      } finally {
        setLoadingAddCartPrime(false);
      }
    }), [data, handleOnModalHideSignIn, isPrime, loadingAddCartPrime, navigation, onAddPrimeToCart]);
    var onCloseModalWelcomePrime = (0, _react.useCallback)(function () {
      handleClickContinue();
      setTimeout(function () {
        navigation.navigate('Home');
      }, 300);
    }, [handleClickContinue, navigation]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_$$_REQUIRE(_dependencyMap[17]).SafeAreaView, {
      style: {
        backgroundColor: '#fff'
      },
      children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsxs)(_$$_REQUIRE(_dependencyMap[18]).Box, {
        bg: "white",
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_$$_REQUIRE(_dependencyMap[19]).TopBarDefaultBackButton, {
          loading: loading || loadingAddCartPrime,
          navigateGoBack: true
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_$$_REQUIRE(_dependencyMap[20]).ModalBag, {
          isVisible: animationBag,
          onBackdropPress: function onBackdropPress() {
            return changeStateAnimationBag(false);
          },
          onModalHide: handleOnModalHide
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_$$_REQUIRE(_dependencyMap[21]).ModalWelcomePrime, {
          isVisible: isVisibleModalWelcome,
          onClose: onCloseModalWelcomePrime
        }), !!(data && !loading) && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsxs)(_reactNative.ScrollView, Object.assign({
          contentContainerStyle: {
            paddingBottom: 100
          }
        }, (0, _testProps.default)('PrimeLP_page'), {
          children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeHero.default, {
            data: data,
            onAddToCart: onAddToCart
          }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeIntro.default, {
            data: data
          }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeBenefits.default, {
            data: data
          }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeSubscribe.default, {
            data: data,
            onAddToCart: onAddToCart
          }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeFAQ.default, {
            data: data
          })]
        }))]
      })
    });
  }
  var _default = exports.default = PrimeLP;
