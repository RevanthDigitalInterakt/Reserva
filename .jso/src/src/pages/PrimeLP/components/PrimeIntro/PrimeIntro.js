  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _react = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[2]);
  var _testProps = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[3]));
  function PrimeIntro(_ref) {
    var data = _ref.data;
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_reactNative.View, Object.assign({
      style: _$$_REQUIRE(_dependencyMap[5]).styles.wrapper
    }, (0, _testProps.default)('PrimeIntro_wrapper'), {
      children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsxs)(_reactNative.View, {
        style: _$$_REQUIRE(_dependencyMap[5]).styles.elevatedBox,
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsxs)(_$$_REQUIRE(_dependencyMap[6]).Typography, {
          variant: "tituloSessoes",
          style: _$$_REQUIRE(_dependencyMap[5]).styles.title,
          children: ["A experi\xEAncia", /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[6]).Typography, {
            variant: "descontoTag1",
            style: _$$_REQUIRE(_dependencyMap[5]).styles.titleStrong,
            children: ' Reserva'
          }), '\n', "em seu modo", /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[6]).Typography, {
            variant: "descontoTag1",
            style: _$$_REQUIRE(_dependencyMap[5]).styles.titleStrong,
            children: ' Premium'
          })]
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[6]).Typography, {
          variant: "tituloSessao",
          style: _$$_REQUIRE(_dependencyMap[5]).styles.subtitle,
          children: "Fa\xE7a parte do Prime, o clube de vantagens da Reserva, e tenha acesso a benef\xEDcios em todos os produtos da marca, al\xE9m de experi\xEAncias exclusivas em parceiros que j\xE1 completam seu estilo de vida."
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsxs)(_$$_REQUIRE(_dependencyMap[6]).Typography, {
          variant: "tituloSessao",
          style: _$$_REQUIRE(_dependencyMap[5]).styles.subtitle,
          children: ["Assine em at\xE9", /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[6]).Typography, {
            variant: "precoTotal",
            style: _$$_REQUIRE(_dependencyMap[5]).styles.subtitleStrong,
            children: ` ${data.installmentQty}x de R$ ${data.installmentPrice} `
          }), "e comece a aproveitar agora! Depois que se tornar membro, voc\xEA n\xE3o vai se arrepender."]
        })]
      })
    }));
  }
  var _default = exports.default = PrimeIntro;
