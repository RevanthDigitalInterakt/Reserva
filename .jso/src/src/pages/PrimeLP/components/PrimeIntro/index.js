  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function get() {
      return _PrimeIntro.default;
    }
  });
  var _PrimeIntro = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
