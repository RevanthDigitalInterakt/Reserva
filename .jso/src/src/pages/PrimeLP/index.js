  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function get() {
      return _PrimeLP.default;
    }
  });
  var _PrimeLP = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
