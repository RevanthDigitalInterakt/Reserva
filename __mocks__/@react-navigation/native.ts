const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
