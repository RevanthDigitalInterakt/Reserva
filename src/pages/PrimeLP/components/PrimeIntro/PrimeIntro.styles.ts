import { StyleSheet } from 'react-native';
import { COLORS } from '../../../../base/styles/colors';

export const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: '#F7F7F7',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 32,
  },

  elevatedBox: {
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.23,
    shadowRadius: 2.62,
    elevation: 4,
    backgroundColor: '#ffffff',
    paddingHorizontal: 16,
    paddingTop: 32,
    paddingBottom: 8,
    marginBottom: 32,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: -60,
    width: '100%',
  },

  title: {
    fontSize: 20,
    lineHeight: 26,
    color: '#000000',
    marginBottom: 24,
  },

  titleStrong: {
    fontSize: 20,
    lineHeight: 26,
    color: '#000000',
  },

  subtitle: {
    fontSize: 16,
    lineHeight: 20,
    color: COLORS.GRAY,
    marginBottom: 24,
    textAlign: 'center',
  },

  subtitleStrong: {
    fontSize: 16,
    lineHeight: 20,
    color: '#000000',
  },
});
