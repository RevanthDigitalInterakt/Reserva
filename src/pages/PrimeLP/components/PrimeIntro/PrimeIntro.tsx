import React from 'react';
import { View } from 'react-native';
import { styles } from './PrimeIntro.styles';
import type { PrimeDetailOutput } from '../../../../base/graphql/generated';
import testProps from '../../../../utils/testProps';
import { Typography } from '../../../../components/Typography/Typography';

interface IPrimeIntro {
  data: PrimeDetailOutput;
}

function PrimeIntro({ data }: IPrimeIntro) {
  return (
    <View style={styles.wrapper} {...testProps('PrimeIntro_wrapper')}>
      <View style={styles.elevatedBox}>
        <Typography variant="tituloSessoes" style={styles.title}>
          A experiência
          <Typography variant="descontoTag1" style={styles.titleStrong}>
            {' Reserva'}
          </Typography>
          {'\n'}
          em seu modo
          <Typography variant="descontoTag1" style={styles.titleStrong}>
            {' Premium'}
          </Typography>
        </Typography>

        <Typography variant="tituloSessao" style={styles.subtitle}>
          Faça parte do Prime, o clube de vantagens da Reserva, e tenha acesso
          a benefícios em todos os produtos da marca, além de experiências
          exclusivas em parceiros que já completam seu estilo de vida.
        </Typography>

        <Typography variant="tituloSessao" style={styles.subtitle}>
          Assine em até
          <Typography
            variant="precoTotal"
            style={styles.subtitleStrong}
          >
            {` ${data.installmentQty}x de R$ ${data.installmentPrice} `}
          </Typography>
          e comece a aproveitar agora! Depois que se tornar membro,
          você não vai se arrepender.
        </Typography>
      </View>
    </View>
  );
}

export default PrimeIntro;
