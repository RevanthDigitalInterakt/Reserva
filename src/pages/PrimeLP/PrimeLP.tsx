import React, { useCallback, useMemo, useState } from 'react';
import { ScrollView } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Box } from '@usereservaapp/reserva-ui';
import { TopBarDefaultBackButton } from '../../modules/Menu/components/TopBarDefaultBackButton';
import PrimeHero from './components/PrimeHero';
import PrimeIntro from './components/PrimeIntro';
import PrimeBenefits from './components/PrimeBenefits';
import PrimeSubscribe from './components/PrimeSubscribe';
import { useLandingPagePrimeQuery } from '../../base/graphql/generated';
import { useCart } from '../../context/CartContext';
import { ModalBag } from '../../common/components/ModalBag/ModalBag';
import EventProvider from '../../utils/EventProvider';
import { useApolloFetchPolicyStore } from '../../zustand/useApolloFetchPolicyStore';
import testProps from '../../utils/testProps';

function PrimeLP() {
  const { getFetchPolicyPerKey } = useApolloFetchPolicyStore(['getFetchPolicyPerKey']);
  const [loadingAddCart, setLoadingAddCart] = useState(false);
  const [animationBag, setAnimationBag] = useState(false);

  const { addItem } = useCart();

  const { data: rawData, loading } = useLandingPagePrimeQuery({
    context: { clientName: 'gateway' },
    fetchPolicy: getFetchPolicyPerKey('landingPagePrime'),
  });

  const data = useMemo(() => rawData?.landingPagePrime, [rawData?.landingPagePrime]);

  const onAddPrimeToCart = useCallback(async () => {
    try {
      if (!data || loadingAddCart) return;

      setLoadingAddCart(true);

      await addItem({
        quantity: 1,
        itemId: `${data.productId}`,
        seller: data.productSeller,
      });

      EventProvider.logEvent('add_to_cart_prime', {
        item_quantity: 1,
        item_id: `${data.productId}`,
        seller: data.productSeller,
      });

      setLoadingAddCart(false);
      setAnimationBag(true);
    } catch (e) {
      EventProvider.captureException(e);
    }
  }, [addItem, data, loadingAddCart]);

  return (
    <SafeAreaView style={{ backgroundColor: '#fff' }}>
      <Box bg="white">
        <TopBarDefaultBackButton loading={loading || loadingAddCart} navigateGoBack />

        <ModalBag
          isVisible={animationBag}
          onBackdropPress={() => setAnimationBag(false)}
        />

        {!!(data && !loading) && (
          <ScrollView
            contentContainerStyle={{ paddingBottom: 100 }}
            {...testProps('PrimeLP_page')}
          >
            <PrimeHero data={data} onAddToCart={onAddPrimeToCart} />

            <PrimeIntro data={data} />

            <PrimeBenefits data={data} />

            <PrimeSubscribe data={data} onAddToCart={onAddPrimeToCart} />
          </ScrollView>
        )}
      </Box>
    </SafeAreaView>
  );
}

export default PrimeLP;
