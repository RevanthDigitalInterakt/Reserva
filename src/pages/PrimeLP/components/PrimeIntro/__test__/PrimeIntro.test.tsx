import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '@usereservaapp/reserva-ui';
import { render, screen } from '@testing-library/react-native';

import { mockPrimeData } from '../../../../../../__mocks__/PrimeLP.mock';
import PrimeIntro from '../PrimeIntro';

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <PrimeIntro data={mockPrimeData} />
  </ThemeProvider>
);

describe('PrimeIntro', () => {
  beforeEach(() => {
    render(TestingComponent);
  });

  it('should render properly and match to snapshot', () => {
    const primeIntro = screen.getByTestId('com.usereserva:id/PrimeIntro_wrapper');
    const primeIntroSnapshot = screen.toJSON();

    expect(primeIntro).toBeOnTheScreen();
    expect(primeIntroSnapshot).toMatchSnapshot();
  });
});
