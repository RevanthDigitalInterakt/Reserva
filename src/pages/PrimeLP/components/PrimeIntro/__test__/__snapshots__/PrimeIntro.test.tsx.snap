// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeIntro should render properly and match to snapshot 1`] = `
<View
  accessible={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7F7F7",
      "justifyContent": "center",
      "paddingHorizontal": 32,
    }
  }
  testID="com.usereserva:id/PrimeIntro_wrapper"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "elevation": 4,
        "justifyContent": "center",
        "marginBottom": 32,
        "marginTop": -60,
        "paddingBottom": 8,
        "paddingHorizontal": 16,
        "paddingTop": 32,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 2.62,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ReservaSerif-Regular",
            "fontSize": 28,
          },
          Object {
            "color": "#000000",
            "fontSize": 20,
            "lineHeight": 26,
            "marginBottom": 24,
          },
        ]
      }
      variant="tituloSessoes"
    >
      A experiência
      <Text
        style={
          Array [
            Object {
              "fontFamily": "ReservaSerif-Bold",
              "fontSize": 27,
            },
            Object {
              "color": "#000000",
              "fontSize": 20,
              "lineHeight": 26,
            },
          ]
        }
        variant="descontoTag1"
      >
         Reserva
      </Text>
      

      em seu modo
      <Text
        style={
          Array [
            Object {
              "fontFamily": "ReservaSerif-Bold",
              "fontSize": 27,
            },
            Object {
              "color": "#000000",
              "fontSize": 20,
              "lineHeight": 26,
            },
          ]
        }
        variant="descontoTag1"
      >
         Premium
      </Text>
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
          },
          Object {
            "color": "#848484",
            "fontSize": 12,
            "lineHeight": 15.5,
            "marginBottom": 24,
            "textAlign": "center",
          },
        ]
      }
      variant="tituloSessao"
    >
      Faça parte do Prime, o clube de vantagens da Reserva, e tenha acesso a benefícios em todos os produtos da marca, além de experiências exclusivas em parceiros que já completam seu estilo de vida.
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
          },
          Object {
            "color": "#848484",
            "fontSize": 12,
            "lineHeight": 15.5,
            "marginBottom": 24,
            "textAlign": "center",
          },
        ]
      }
      variant="tituloSessao"
    >
      Assine em até
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Nunito-Bold",
              "fontSize": 20,
            },
            Object {
              "color": "#000000",
              "fontSize": 12,
              "lineHeight": 15.5,
            },
          ]
        }
        variant="precoTotal"
      >
         12x de R$ 25 
      </Text>
      e comece a aproveitar agora! Depois que se tornar membro, você não vai se arrepender.
    </Text>
  </View>
</View>
`;
